crates/sound/src/vhu_sound.rs

```diff
@@ -275,34 +286,38 @@ impl VhostUserSoundThread {
                         .memory()
                         .read_obj::<VirtioSndPcmSetParams>(desc_request.addr())
                         .map_err(|_| Error::DescriptorReadFailed)?;
-                    let stream_id: usize = u32::from(set_params.hdr.stream_id) as usize;
+
+                    let params = PCMParams {
+                        buffer_bytes :  set_params.buffer_bytes.into(),
+                        period_bytes : set_params.period_bytes.into(),
+                        features : set_params.features.into(),
+                        rate : set_params.rate.into(),
+                        format : set_params.format.into(),
+                        channels : set_params.channels.into()
+                    };
+
+                    let stream_id = u32::from(set_params.hdr.stream_id);
                     let buffer_bytes: u32 = set_params.buffer_bytes.into();
```

Comment:
maybe we should remove this, since it's already defined above

Reply:
I will do that thanks.